mesonbuild/environment.py

```diff
@@ -624,11 +611,11 @@ def sanitize(p):
                 if version == 'unknown version':
                     version = search_version(out)
                 if version == 'unknown version':
-                    m = 'Failed to detect MSVC compiler arch: stderr was\n{!r}'
+                    m = 'Failed to detect MSVC compiler version: stderr was\n{!r}'
                     raise EnvironmentException(m.format(err))
-                is_64 = err.split('\n')[0].endswith(' x64')
+                arch = err.split('\n')[0].rsplit(' ', 1)[1].strip()
```

Comment:
This is far too brittle. Compiler older than an certain vintage (2005?) possibly don't identify the target, so we should default to x86 (as we currently do) in that case.